File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: true

class ResistorColorDuo
  BANDS = %w[black brown red orange yellow green blue violet grey white].freeze

  def self.value(codes, *_args)
    first, second = *codes
    "#{BANDS.index(first)}#{BANDS.index(second)}".to_i
  end
end

File: python/resistor-color-duo/resistor_color_duo.py
def value(colors):
    first, second, *_ = colors
    bands = [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    ]

    return int(f'{bands.index(first)}{bands.index(second)}')
